src/ejs/includes/app.ejs

```diff
@@ -0,0 +1,142 @@
+              @click="changeFiat(currency.name, currency.ratio)"
+            >
+              {{ currency.name }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div class="App-address">
+      <input
+        class="App-address-input form-control"
+        type="text"
+        v-model="btcaddress"
+        placeholder="Bitcoin address"
+        required
+      />
+
+      <div class="App-address-extra">
+        <div class="App-address-extra-error" v-show="showAddressError">
+          {{ getAddressError }}
+        </div>
+
+        <div class="App-address-extra-get">
+          <a target="_blank" :href="config.walleturl">Don't have one?</a>
+        </div>
+      </div>
+    </div>
+
+    <div class="App-captcha">
+      <div ref="rehead" :class="showRecaptchaError? 'recaptcha_box_error' : ''">
+        <div
+          :class="config.recaptcha_site_key? 'g-recaptcha' : ''"
+          :data-sitekey="config.recaptcha_site_key"
+        ></div>
+      </div>
+    </div>
+
+    <div class="App-submit">
+      <button type="submit" class="App-submit-btn">
+        Continue
+      </button>
+    </div>
+  </form>
+
+  <div ref="formplaceholder"></div>
+</div>
+<%# the control end %>
+
+<%# control JS %>
+<script src="https://www.google.com/recaptcha/api.js"></script>
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+<script src="https://unpkg.com/vue/dist/vue.js"></script>
+<script type="text/javascript" src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-alpha.5/js/bootstrap.min.js"></script>
+<script src="./javascripts/sha256.js"></script>
+<script src="./javascripts/bigint.js"></script>
+<script src="./javascripts/btc_address_validator.js"></script>
+
+<script>
+  var i_lastQuote = {};
+  i_lastQuote.user_id = 'quote.user_id ';
+  i_lastQuote.quote_id = 'quote.quote_id';
+  i_lastQuote.fiat_money = {};
+  i_lastQuote.fiat_money.currency = 'USD';
+  i_lastQuote.fiat_money.amount = 120;
+  i_lastQuote.digital_money = {};
+  i_lastQuote.digital_money.currency = 'BTC';
+  i_lastQuote.digital_money.amount = 0.16911889;
+  i_lastQuote.base_rate = 700;
+
+  var config = {};
+  config.lastQuote = i_lastQuote;
+  config.csrfToken = 'csrfTokenKey';
+  config.recaptcha_site_key= '6LdCLyITAAAAAO7JocwJAmhDYRpKpQHyJqnnMsnT';
+  config.walleturl= 'https://www.simplex.com';
+</script>
+
+<script type="text/javascript" src="./javascripts/currencyconvertor.js"></script>
+<%# control JS end %>
+
+<%# custom JS begin %>
+<script>
+  // Pull address out of query, stick in input
+  (function() {
+    const queryMatches = window.location.href.match(/[?&]address=([0-9a-zA-Z]+)/i);
+    if (queryMatches && queryMatches[1]) {
+      const addrInput = document.querySelector('.App-address-input');
+      addrInput.value = queryMatches[1];
```

Comment:
any possibility this could lead to xss?

Reply:
Nope - the possible values are alphanumeric as per the `match` on line 135

Reply:
@409H has it, this only picks up `0-9a-zA-Z`. I was very concerned about XSS after all of our recent struggles, haha.